### ISSUE TITLE
SVG text elements do not support textLength and lengthAdjust attributes

### ISSUE THREAD

**Issue by oshmoun** [Kozea/WeasyPrint#1921]
According to [the specs](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/textLength), `textLength` should be a valid attribute for SVG text elements. However, it seems Weasyprint is ignoring it. The same applies for `lengthAdjust`.

Here's an example SVG:
```svg
<svg width="240" height="40" viewBox="0 0 240 40" xmlns="http://www.w3.org/2000/svg">
<text fill="black"><tspan x="10" y="20" textLength="20" lengthAdjust="spacingAndGlyphs">short</tspan> <tspan textLength="200" lengthAdjust="spacingAndGlyphs">long</tspan></text>
</svg>
```

Expected:
![image](https://github.com/Kozea/WeasyPrint/assets/2135622/8e9f39c6-927d-46b2-add8-aa34843a5bb9)

Result:
![image](https://github.com/Kozea/WeasyPrint/assets/2135622/23a0a889-deec-44d0-9ffd-71195a713592)

**Comment by liZe** [comment]
That would be a nice feature to have!

**Comment by oshmoun** [comment]
Implemented the PoC, here's the result for the SVG in the description:
![image](https://github.com/Kozea/WeasyPrint/assets/2135622/ef4f4b25-d622-4d1d-a604-8408460ccd02)

while it works quite nicely, the spacing strategy has the unintended consequence of introducing whitespace within the words, making them unsearchable. Not sure how that could be avoided :sweat:

**Comment by oshmoun** [comment]
Ok I ended up also implementing `spacingAndGlyphs`, which also avoids the whitespace issue observed in the previous comment :partying_face: 

works pretty well!
![image](https://github.com/Kozea/WeasyPrint/assets/2135622/22a3489c-63d1-4ae9-ae8b-e64e60c4828e)

**Comment by liZe** [comment]
That’s great, thanks a lot! 😍💜

I’ll clean a few things and get back to you soon.

If you have some motivation left, you can add some tests in [`test_text.py`](https://github.com/Kozea/WeasyPrint/blob/master/tests/draw/svg/test_text.py). Other tests in this file should be enough for you to understand how they work 😄. You’ll find extra tips about how to launch tests in [the doc](https://doc.courtbouillon.org/weasyprint/stable/contribute.html) if needed! (And of course, I can add the tests myself if you don’t want to, no problem at all.)

**Comment by oshmoun** [comment]
> If you have some motivation left, you can add some tests in [`test_text.py`](https://github.com/Kozea/WeasyPrint/blob/master/tests/draw/svg/test_text.py). Other tests in this file should be enough for you to understand how they work 😄. You’ll find extra tips about how to launch tests in [the doc](https://doc.courtbouillon.org/weasyprint/stable/contribute.html) if needed! (And of course, I can add the tests myself if you don’t want to, no problem at all.)

I'll look into the tests today, and push them to the PR when ready :smiley:

**Comment by liZe** [comment]
Fixed by #1922.
